plugins/scanners/fossid/src/main/kotlin/FossId.kt

```diff
@@ -236,110 +236,111 @@ class FossId internal constructor(
-                    e.showStackTrace()
+                val (scanCode, scanId) = if (config.deltaScans) {
+                    checkAndCreateDeltaScan(scans, url, revision, projectCode, projectName, context, issues)
+                } else {
+                    checkAndCreateScan(scans, url, revision, projectCode, projectName)
+                }
 
+                if (config.waitForResult) {
+                    val rawResults = getRawResults(scanCode)
+                    createResultSummary(
+                        startTime,
+                        provenance,
+                        rawResults,
+                        scanCode,
+                        scanId,
+                        issues,
+                        context.detectedLicenseMapping
+                    )
+                } else {
                     val issue = createAndLogIssue(
                         source = name,
-                        message = "Failed to scan package '${pkg.id.toCoordinates()}' from $url."
+                        message = "Package '${pkg.id.toCoordinates()}' has been scanned in asynchronous mode. " +
+                            "Scan results need to be inspected on the server instance.",
+                        severity = Severity.HINT
                     )
                     val summary = createSingleIssueSummary(startTime, issue = issue)
 
-                    if (!config.keepFailedScans) {
-                        createdScans.forEach { code ->
-                            logger.warn { "Deleting scan '$code' during exception cleanup." }
-                            deleteScan(code)
-                        }
-                    }
+                    ScanResult(
+                        provenance,
+                        details,
+                        summary,
+                        mapOf(SCAN_CODE_KEY to scanCode, SCAN_ID_KEY to scanId, SERVER_URL_KEY to config.serverUrl)
+                    )
+                }
+            } catch (e: IllegalStateException) {
+                e.showStackTrace()
 
-                    ScanResult(provenance, details, summary)
+                val issue = createAndLogIssue(
+                    source = name,
+                    message = "Failed to scan package '${pkg.id.toCoordinates()}' from $url."
+                )
+                val summary = createSingleIssueSummary(startTime, issue = issue)
+
+                if (!config.keepFailedScans) {
+                    createdScans.forEach { code ->
+                        logger.warn { "Deleting scan '$code' during exception cleanup." }
+                        deleteScan(code)
+                    }
                 }
+
+                ScanResult(provenance, details, summary)
             }
         }
 
-        logger.info { "Scan has been performed. Total time was $duration." }
+        logger.info {
+            val duration = with(result.summary) { java.time.Duration.between(startTime, endTime).toKotlinDuration() }
```

Comment:
Can't you just do `val duration = startTime - endTime`? (https://kotlinlang.org/api/kotlinx-datetime/kotlinx-datetime/kotlinx.datetime/-instant/#126222255%2FFunctions%2F-1003227268)

Reply:
Ok, I forgot that we use Java instants in the model...

Reply:
So you could do `endTime.toKotlinInstant() - startTime.toKotlinInstant()` but that's not better.

Reply:
And `toKotlinInstant()` is only available with *kotlinx-datetime*, I believe.